TutorTask97_Spring2025_Real-time_Bitcoin_Analysis_with_Langchain_and_Neo4j

vineetj5 opened the pull request:
Added tutorial template files
Github Issue - https://github.com/causify-ai/tutorials/issues/97

tkpratardan reviewed (changes requested):
Good job starting the project!

I would recommend copying the new set of template files to reflect the updated changes for using docker. Work on this branch for some commits, push your progress and we will take a checkpoint after that.

vineetj5 commented:
These are the new changes which I have worked on: 
- Added Dockerfile and docker-compose.yml for orchestrating Neo4j with APOC and a Python loader service.
- Created requirements.txt to manage all Python dependencies for both local and containerized execution.
- Developed load_backup_to_db.py to automatically load transactions from bitcoin_transactions_backup.json into Neo4j on container startup.
- Added neo4j and the loader services in the docker-compose.yml.
- Implemented healthcheck and persistent volume for Neo4j in docker-compose.yml.
- Created config.py to use Docker service hostname for Neo4j connectivity and also the Blockstream API URLs for fetching the Transaction details.
- Implemented insert_transactions which would create nodes and establish relationships based on the Transaction data.
- Ensured all HTTP requests use retry logic to prevent network-related failures.
- Added Jupyter notebooks (prevDataLoad.ipynb, realTimeDataIngestion.ipynb) for data exploration and real-time ingestion.
- Updated and maintained transaction data files (bitcoin_confirmed_transactions.jsonl, bitcoin_transactions_backup.json, realtime_bitcoin_transactions.jsonl).
- Added template_utils.py and docker_data605_style/ for utility and style management.
- Added README_tutorTask97.md which has all the details of how to run the project.
